spacy_llm/models/integration/langchain.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, Callable, Dict, Iterable, Optional, Type
+
+from spacy.util import SimpleFrozenDict
+
+from ...compat import has_langchain, langchain
+from ...registry import registry
+from . import Remote
+
+
+def _check_installation() -> None:
+    """Checks whether `langchain` is installed. Raises an error otherwise."""
+    if not has_langchain:
+        raise ValueError(
+            "The LangChain model requires `langchain` to be installed, which it is not. See "
+            "https://github.com/hwchase17/langchain for installation instructions."
+        )
+
+
+@registry.llm_queries("spacy.CallLangChain.v1")
+def query_langchain() -> (
+    Callable[["langchain.llms.base.BaseLLM", Iterable[Any]], Iterable[Any]]
+):
+    """Returns query Callable for LangChain.
+    RETURNS (Callable[["langchain.llms.BaseLLM", Iterable[Any]], Iterable[Any]]:): Callable executing simple prompts on
+        the specified LangChain model.
+    """
+    return _prompt_langchain
+
+
+def _prompt_langchain(
+    model: "langchain.llms.base.BaseLLM", prompts: Iterable[Any]
+) -> Iterable[Any]:
+    return [model(pr) for pr in prompts]
+
+
+def register_langchain_models() -> None:
+    """Registers APIs supported by langchain (one API is registered as one model)."""
+    type_to_cls_dict: Dict[
+        str, Type[langchain.llms.base.BaseLLM]
+    ] = langchain.llms.type_to_cls_dict
+
+    for class_id, cls in type_to_cls_dict.items():
+
+        def langchain_model(
+            query: Optional[
+                Callable[["langchain.llms.base.BaseLLM", Iterable[str]], Iterable[str]]
+            ] = None,
+            config: Dict[Any, Any] = SimpleFrozenDict(),
+            langchain_class_id: str = class_id,
+        ) -> Callable[[Iterable[Any]], Iterable[Any]]:
+            model = config.pop("model")
+            try:
+                return Remote(
+                    integration=type_to_cls_dict[langchain_class_id](
+                        model_name=model, **config
+                    ),
+                    query=query_langchain() if query is None else query,
+                )
+            except ImportError as ex:
+                raise ValueError(
+                    "Failed to instantiate LangChain class. Ensure all necessary dependencies are installed."
+                ) from ex
```

Comment:
This probably should error/exit, but warn instead? Because you might just want to use `spacy-llm` without LangChain but then if this breaks, you're blocked.

Reply:
`register_langchain_models()` is only called [if `langchain` is available](https://github.com/explosion/spacy-llm/pull/176/files#diff-b461f0886b68b6bbf5bda73e7feebf4f6667fc7daa58a14cb4b9a62cb9b4196cR70).

Reply:
I realise that, but even then. Imagine having LangChain in your environment, but you're not actually using it, and these register commands break. Then you can't continue with any functionality in `spacy-llm`. In general, we should really only error if we're 100% sure that there's no further usage possible.

Reply:
Fair point. Warning instead of raising in https://github.com/explosion/spacy-llm/pull/176/commits/05c4ee2e4ca4efdc9f2a62f5f70038dbdc4c4d3a.